components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41Utils.java

```diff
@@ -3170,7 +3170,27 @@ public static void deactivateTenant(String tenantDomain) throws RestAPIException
         int tenantId;
         try {
             tenantId = tenantManager.getTenantId(tenantDomain);
+            if (tenantId != -1) {
+                try {
+                    TenantMgtUtil.deactivateTenant(tenantDomain, tenantManager, tenantId);
+                } catch (Exception e) {
+                    String msg = "Error in deactivating Tenant :" + tenantDomain;
+                    log.error(msg, e);
+                    throw new RestAPIException(msg, e);
+                }
 
+                //Notify tenant deactivation all listeners
+                try {
+                    TenantMgtUtil.triggerTenantDeactivation(tenantId);
+                } catch (StratosException e) {
+                    String msg = "Error in notifying tenant deactivate.";
+                    log.error(msg, e);
+                    throw new RestAPIException(msg, e);
+                }
+            } else {
+                String msg = "The tenant with domain name: " + tenantDomain + " does not exist.";
+                throw new RestAPIException(msg);
```

Comment:
Is this a specific exception? Or an internal issue? 
If it is not an internal issue, we should not throw RestAPIException.